packages/plugins/typescript-client/src/helpers.ts

```diff
@@ -10,3 +16,74 @@ export function shouldHavePrefix(type: Field, options: Handlebars.HelperOptions)
 export function isPrimitiveType(type: Field, options: Handlebars.HelperOptions) {
   return options.data.root.primitives[type.type];
 }
+
+function nameFragment(
+  convert: (str: string) => string,
+  prefix: string,
+  fragment: SelectionSetFragmentSpread | SelectionSetInlineFragment,
+  noNamespaces: boolean
+) {
+  if (isFragmentSpread(fragment)) {
+    return convert(fragment.fragmentName) + (noNamespaces ? '' : '.') + 'Fragment';
+  }
+
+  return (noNamespaces ? convert(prefix) : '') + fragment.onType;
+}
+
+function isFragmentSpread(
+  fragment: SelectionSetFragmentSpread | SelectionSetInlineFragment
+): fragment is SelectionSetFragmentSpread {
+  return fragment.isFragmentSpread;
+}
+
+export function fragments(convert: (str: string) => string) {
+  return (operation: Operation, prefix: string, fragments: Fragment[], options: Handlebars.HelperOptions): string => {
+    const config = options.data.root.config || {};
+    const noNamespaces = config.noNamespaces === true;
+    const fragmentsByType: {
+      [type: string]: Array<string>;
+    } = {};
+
+    operation.inlineFragments.forEach(fragment => {
+      // FIX: somehow `fragment.onType` has `InlineFragment` suffix
+      const type = fragment.onType.replace('InlineFragment', '');
```

Comment:
@dotansimha this one is super weird. I'm looking to find a source of that.

Reply:
It's done by `flattenTypes`. Look for the function `buildModelFromInlineFragment`.
@kamilkisiela